### ISSUE TITLE
Todo regex seemingly only searches in commented text, and structure does not update automatically

### ISSUE THREAD

**Issue by john-blip** [texstudio-org/texstudio#3545]
### Environment
- TeXstudio: 4.7.2
- Qt: 6.6.1
- OS: Windows 10
- TeX distribution: texlive

### Expected behavior
**1: Todo regex seemingly only searches in commented text** 
I'm using custom commands for todo-ing, named after myself and supervisors, to indicate who made them (e.g. `\testname{comment here}`). I defined a regex to search for all these commands, to have them recognised as 'to do' note to show in the left-hand structure panel.
  
Similarly, I have copied these commands and appended 'resolved' (e.g. `\testnameResolved{comment here}`) to indicate addressed comments. These should not be included.

I include both command types in 'regular' text; not in comments, as is the case in the highlighting of eg `% TODO`, so I dropped the % from the standard regex (see eg the [manual](https://texstudio-org.github.io/configuration.html#configuring-the-advanced-editor-settings-advanced-option)).
I have defined the following regex in the advanced editor: `(\\testname)[^Resolved]`

**2: Todo structure does not update automatically** 
I also expect the structure to be updated automatically, or at least upon recompiling or when using Idefix > Refresh structure. 

### Actual behavior
It appears that the regex only searches in commented-out lines. See the MWE screenshots below: lines 2, 5, and 6 are working as expected; line 3 is not as it is not included in the TO DO overview in structure.
![image](https://github.com/texstudio-org/texstudio/assets/71980492/cbe3542a-541d-4a36-88ae-1c55b158db28)
![image](https://github.com/texstudio-org/texstudio/assets/71980492/82deae15-41de-4c43-8719-c7513f41d544)

**2: Todo structure does not update automatically** 
Changes made (i.e. adding a `\testname{}`) after Texstudio is started, aren't included in the structure until after rebooting Texstudio.

**Comment by octaeder** [comment]
Indeed todo comments need to start with a % (that's why they are called todo comment). In fact using `TODO|todo` in the Adv. Edit option field for the rexexp does the same as the default setting `%\s*(TODO|todo)`. The manual uses exactly the same wording.

![image](https://github.com/texstudio-org/texstudio/assets/102688820/0272f1a9-242a-440b-9de4-9baeea9a50dc)

@sunderme Maybe this should be labeled as an enhancement.

**Comment by john-blip** [comment]
Ah, I see, thanks for your reply.
I figured that because the `%` is included in the example regex in the manual, removing that would make Texstudio look for any command/text that satisfies the regex and that the users uses to write 'comments' (so interpreting the 'comment' part in the setting not as 'commented-out-text' but as 'a comment regarding the text').
If the scope cannot be expanded to also include non-commented text, perhaps this could be mentioned explicitly in the manual?

**Comment by sunderme** [comment]
no, to use command arguments as todo comments, cwl needs to be used.
See https://texstudio-org.github.io/background.html#classification-format ('D')
